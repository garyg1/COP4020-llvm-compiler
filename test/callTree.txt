extern int print_float(float x);
extern int print_add(float a, float b);

def float add(float a, float b) (
    print_add(a, b):
    a + b
);
def float addOne(float a) a + 1.0;
def float plusOneTimesTwo(float a) add(addOne(a), addOne(a));
def float twice(float a) add(plusOneTimesTwo(a), plusOneTimesTwo(a));
def float thrice(float a) add(twice(a), twice(a));


def int main() (
    print_float(thrice(10.0))
);