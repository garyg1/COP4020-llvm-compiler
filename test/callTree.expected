Adding 11.000000 + 11.000000
Adding 11.000000 + 11.000000
Adding 22.000000 + 22.000000
Adding 11.000000 + 11.000000
Adding 11.000000 + 11.000000
Adding 22.000000 + 22.000000
Adding 44.000000 + 44.000000
88.000000
